ksqldb-parser/src/main/java/io/confluent/ksql/util/QueryAnonymizer.java

```diff
@@ -0,0 +1,776 @@
+import io.confluent.ksql.parser.SqlBaseParser.DropTableContext;
+import io.confluent.ksql.parser.SqlBaseParser.DropTypeContext;
+import io.confluent.ksql.parser.SqlBaseParser.ExplainContext;
+import io.confluent.ksql.parser.SqlBaseParser.ExpressionContext;
+import io.confluent.ksql.parser.SqlBaseParser.GroupByContext;
+import io.confluent.ksql.parser.SqlBaseParser.InsertIntoContext;
+import io.confluent.ksql.parser.SqlBaseParser.InsertValuesContext;
+import io.confluent.ksql.parser.SqlBaseParser.IntegerLiteralContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListConnectorsContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListFunctionsContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListPropertiesContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListQueriesContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListStreamsContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListTopicsContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListTypesContext;
+import io.confluent.ksql.parser.SqlBaseParser.ListVariablesContext;
+import io.confluent.ksql.parser.SqlBaseParser.LogicalBinaryContext;
+import io.confluent.ksql.parser.SqlBaseParser.PartitionByContext;
+import io.confluent.ksql.parser.SqlBaseParser.PrintTopicContext;
+import io.confluent.ksql.parser.SqlBaseParser.QueryContext;
+import io.confluent.ksql.parser.SqlBaseParser.RegisterTypeContext;
+import io.confluent.ksql.parser.SqlBaseParser.SelectItemContext;
+import io.confluent.ksql.parser.SqlBaseParser.SelectSingleContext;
+import io.confluent.ksql.parser.SqlBaseParser.SetPropertyContext;
+import io.confluent.ksql.parser.SqlBaseParser.ShowColumnsContext;
+import io.confluent.ksql.parser.SqlBaseParser.SingleExpressionContext;
+import io.confluent.ksql.parser.SqlBaseParser.SingleStatementContext;
+import io.confluent.ksql.parser.SqlBaseParser.StatementsContext;
+import io.confluent.ksql.parser.SqlBaseParser.StringLiteralContext;
+import io.confluent.ksql.parser.SqlBaseParser.TableElementContext;
+import io.confluent.ksql.parser.SqlBaseParser.TableElementsContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertiesContext;
+import io.confluent.ksql.parser.SqlBaseParser.TablePropertyContext;
+import io.confluent.ksql.parser.SqlBaseParser.TerminateQueryContext;
+import io.confluent.ksql.parser.SqlBaseParser.TypeContext;
+import io.confluent.ksql.parser.SqlBaseParser.UndefineVariableContext;
+import io.confluent.ksql.parser.SqlBaseParser.UnquotedIdentifierContext;
+import io.confluent.ksql.parser.SqlBaseParser.UnsetPropertyContext;
+import io.confluent.ksql.parser.SqlBaseParser.ValueExpressionContext;
+import java.util.ArrayList;
+import java.util.Hashtable;
+import java.util.List;
+import java.util.stream.Collectors;
+import org.antlr.v4.runtime.ParserRuleContext;
+import org.antlr.v4.runtime.tree.TerminalNode;
+import org.apache.commons.lang3.StringUtils;
+
+public class QueryAnonymizer {
+
+  public String anonymize(final String query) {
```

Comment:
Can you add an overload that also anonymizes from `ParserRuleContext` - just so that if we use it in situ we have a choice of passing the string or parseTree